assets/components/IconApp.vue

```diff
@@ -0,0 +1,48 @@
+<template>
+    <div>
+        <button :class="['icons', 'cursor-pointer']" :style="{ backgroundColor: color }" @click="handleClick">
+            <font-awesome-icon :icon="iconClass" size="2x" />
+        </button>
+    </div>
+</template>
+
+<script setup lang="ts">
+const props = defineProps({
+    icon: {
+        type: String,
+        required: true,
+    },
+    color: {
+        type: String,
+        default: '#C2C1FF',
+    },
+});
+
+const iconClass = `fa-${props.icon}`;
+
+const handleClick = (): void => {
+    console.log('Hello world');
+};
+</script>
+
+<style lang="css" scoped>
+.icons {
+    color: #4c324d;
+    background-color: var(--color);
+    width: 72px;
+    height: 72px;
+    padding: 8px 8px;
+    font-size: 25px;
+    border-radius: 4px;
+    font-weight: 700;
+    border: 4px solid #4c324d;
+    box-shadow: 8px 8px #4c324d;
+    border-radius: 4px;
```

Comment:
Please use multiples of 0.25em for sizes (and their corresponding classes like p-1, p-2, m-4, etc...)